### PR TITLE
Deep linking documentation

### PR DESCRIPTION
**Pull request by theodur**
## Description of Change

This PR adds documentation for setting up deep linking for push notifications. This should give developers the information they need to configure deep linking for push notifications that they add to the app.  

## Screenshots/Video

N/A

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Overview
+
+Deep links in the app are handled by the React Navigation library. The React Navigation linking configuration handles mapping links to the appropriate screens, and updating the navigation state when necessary.
+
+## Setting up push notifications
+
+A prerequisite for adding deep linking support for a feature's push notification is to have push notifications setup for that feature. An architectural overview of how push notifications work in the app and information on how to add them for a feature can be found [here](../../BackEnd/Features/PushNotifications.md).
+
+In order for deep linking to work for your push notification, the `url` field must be present in the push template. The URL should be prefixed with `vamobile://` and followed by the path of the deep link you want to add, e.g. `vamobile://messages/%MSG_ID%`. In this example, the path of the deep link is prefixed with `vamobile://`, starts with `messages`, and is followed by the `MSG_ID` parameter, which is a field that will be populated from the API call that triggers the push notification. This will result in push notifications returning a payload that looks some like:
+
```
Comment (dumathane) [inline VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md:16]: 'looks some like:' maybe 'looks something like:' ?

**Review by dumathane — COMMENTED** [review]
Looks good, can we add any details on how you send the test pushes during development? There was a way via terminal?

File: VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md (hunk 1 above)
Comment (theodur) [reply VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md:16]: Good catch, just fixed this typo

File: VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+## Handling back navigation
+
+ Once you've added your deep link to the linking configuration, depending on how nested the screen is in the app, one issue you may run into is navigating backwards. Navigating back from screens opened via a deep link will typically go back to the home screen. This is because React Navigation isn't able to automatically generate the correct navigation state when a deep link is opened. The navigation state will need to be manually generated to ensure navigating back after opening a deep link takes the user to the correct screen.
+
+In the linking configuration, the [getStateFromPath](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/8daf0536ebbf801de0ed63e0b2af9385d54b1bc1/VAMobile/src/constants/linking.tsx#L27-L49) function can be used for manually setting the navigation state based on the path of opened the deep link. Logic added here shouldn't conflict with logic for other deep links. The generated navigation state for `ViewMessageScreen` from the previous examples looks like this:
```
[the remark this answers is not in the document]
Comment (theodur) [reply VAMobile/documentation/docs/Engineering/FrontEnd/PushNotifications/DeepLinking.md]: Just fixed this typo also

**Comment by theodur** [comment]
> Looks good, can we add any details on how you send the test pushes during development? There was a way via terminal?

That's a good point, I didn't add any info about how to test deep links locally. This can be done via the `uri-scheme` CLI tool, or with a Detox test. I'll add that in the documentation

**Review by dumathane — APPROVED** [review]

**Review by narin — APPROVED** [review]
Looks good.  Thanks for taking the time to write this up @theodur

**Pull request merged**